src/api.ts

```diff
@@ -72,9 +72,11 @@ router.post('/flag', verifyAuth, async (req, res) => {
   const { type, value, action } = req.body;
 
   try {
-    await flagEntity({ type, value, action });
+    const resp = await flagEntity({ type, value, action });
+    if (!resp.affectedRows) throw new Error('no rows affected');
```

Comment:
Throwing here will make the endpoint return a 500 error. 

Is it really what we want, or should we just use `return res.json({ success: false });` instead, and keep a 200 error code ?

This will let caller distinguish between a genuine server error, from a data error (given ID does not exist)

Reply:
```suggestion
```